### PR TITLE
chore(v2): fix int/float parameter parsing with extra newline and fix reused_component sample

### PR DESCRIPTION
**Pull request by Bobgy**
**Description of your changes:**
Changes:
* when int/float parameter has an extra newline, we can ignore it when parsing its value
* this fixes reused_component sample

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: v2/cmd/launch/main.go [hunk 1]
```diff
@@ -49,10 +49,10 @@ func main() {
 	}
 	launcher, err := component.NewLauncher(*runtimeInfoJSON, opts)
 	if err != nil {
-		glog.Fatalf("Failed to create component launcher: %v", err)
+		glog.Exitf("Failed to create component launcher: %v", err)
```
Comment (Bobgy) [inline v2/cmd/launch/main.go:52]: nit: these stacktraces are useless, so use Exitf instead

File: v2/component/launcher.go [hunk 2]
```diff
@@ -388,27 +388,31 @@ func (l *Launcher) RunComponent(ctx context.Context, cmd string, args ...string)
 	}
 
 	for n, op := range l.runtimeInfo.OutputParameters {
+		msg := func(err error) error {
```
Comment (capri-xiyue) [inline v2/component/launcher.go:391]: Can we rename it to errMsg?

**Comment by neuromage** [comment]
/lgtm

Thanks for the fix!

**Comment by Bobgy** [comment]
/approve

Thank you for the review, I'll get the fix in first. Will address comments after vacation

**Pull request merged**